core/src/main/scala/org/apache/spark/deploy/history/EventLogFileCompactor.scala

```diff
@@ -221,5 +222,19 @@ private class CompactedEventLogFileWriter(
     hadoopConf: Configuration)
   extends SingleEventLogFileWriter(appId, appAttemptId, logBaseDir, sparkConf, hadoopConf) {
 
-  override val logPath: String = originalFilePath.toUri.toString + EventLogFileWriter.COMPACTED
+  override val shouldCompress = EventLogFileWriter.codecName(originalFilePath) match {
+    case Some(_) => true
+    case _ => false
+  }
+
+  override val compressionCodec = {
+    val originalCodecName = EventLogFileWriter.codecName(originalFilePath).getOrElse("")
```

Comment:
Cloud we move line 231 into `if (shouldCompress) {` branch?

Reply:
> Cloud we move line 231 into `if (shouldCompress) {` branch?

has been fixed.